onadata/apps/api/viewsets/export_viewset.py

```diff
@@ -38,7 +40,7 @@ class ExportViewSet(DestroyModelMixin, ReadOnlyModelViewSet):
     permission_classes = [ExportDjangoObjectPermission]
 
     def retrieve(self, request, *args, **kwargs):
-        export = self.get_object()
+        export = get_object_or_404(self.queryset, pk=kwargs.get('pk'))
```

Comment:
So my decision was guided by two things:

1.  In DRF's documents, their retrieve method example uses get_object_or_404, here: http://www.django-rest-framework.org/api-guide/viewsets/

2.  I wasn't able to get self.get_object() to work in my test.  It consistently failed whereas get_object_or_404 just worked